### PR TITLE
Translate Typechecking with Proptypes page

### PR DESCRIPTION
**Pull request by lucasdeassis**
#1

### CODE REVIEW

File: content/docs/typechecking-with-proptypes.md [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-title: Typechecking With PropTypes
+title: Checagem de tipos com PropTypes
 permalink: docs/typechecking-with-proptypes.html
 redirect_from:
   - "docs/react-api.html#typechecking-with-proptypes"
 ---
 
 > Note:
+> 
+> `React.PropTypes` foi movido para um package diferente desde a versão 15.5 do React. Por favor, use [a biblioteca `prop-types`](https://www.npmjs.com/package/prop-types).
```
Comment (klzns) [inline content/docs/typechecking-with-proptypes.md]:
```suggestion
> `React.PropTypes` foi movido para um pacote diferente desde a versão 15.5 do React. Por favor, use [a biblioteca `prop-types`](https://www.npmjs.com/package/prop-types).
```

As suggested by the Style Guide https://github.com/reactjs/pt-BR.reactjs.org/issues/4

File: content/docs/typechecking-with-proptypes.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+> Nós fornecemos [um script de codemod](/react/blog/2017/04/07/react-v15.5.0.html#migrating-from-react.proptypes) para automatizar a conversão.
 
-As your app grows, you can catch a lot of bugs with typechecking. For some applications, you can use JavaScript extensions like [Flow](https://flow.org/) or [TypeScript](https://www.typescriptlang.org/) to typecheck your whole application. But even if you don't use those, React has some built-in typechecking abilities. To run typechecking on the props for a component, you can assign the special `propTypes` property:
+Na medida em que sua aplicação cresce, você pode capturar muitos bugs com checagem de tipos. Em algumas aplicações, você pode usar extensões do JavaScript como [Flow](https://flowtype.org/) ou [TypeScript](https://www.typescriptlang.org/) para checar os tipos de toda a sua aplicação. Porém, mesmo se você não usá-las, React possui algumas habilidades de checagem de tipos embutidas. Para executar a checagem de tipos nas props para um componente, você pode atribuir à propriedade em especial `propTypes`:
```
Comment (klzns) [inline content/docs/typechecking-with-proptypes.md]:
```suggestion
Na medida em que sua aplicação cresce, você pode capturar muitos bugs com checagem de tipos. Em algumas aplicações, você pode usar extensões do JavaScript como [Flow](https://flowtype.org/) ou [TypeScript](https://www.typescriptlang.org/) para checar os tipos de toda a sua aplicação. Porém, mesmo se você não usá-las, React possui algumas habilidades de checagem de tipos nativas. Para executar a checagem de tipos nas props para um componente, você pode atribuir à propriedade em especial `propTypes`:
```

Suggestion based on another Style Guide, but I think it's a good one: https://github.com/cezaraugusto/You-Dont-Know-JS/blob/portuguese-translation/WORDREFERENCE.md#jarg%C3%B5es-de-programa%C3%A7%C3%A3o

File: content/docs/typechecking-with-proptypes.md [hunk 3]
```diff
@@ -168,7 +168,7 @@ ReactDOM.render(
 );
 ```
 
-If you are using a Babel transform like [transform-class-properties](https://babeljs.io/docs/plugins/transform-class-properties/) , you can also declare `defaultProps` as static property within a React component class. This syntax has not yet been finalized though and will require a compilation step to work within a browser. For more information, see the [class fields proposal](https://github.com/tc39/proposal-class-fields).
+Se você está usando um plugin Babel como [plugin-proposal-class-properties](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties), você também poderá declarar `defaultProps` como propriedade estática dentro de uma classe React.Component. Essa sintaxe ainda não foi finalizada e irá exigir uma etapa de compilação para funcionar em um navegador. Para mais informações, veja [proposal-class-fields](https://github.com/tc39/proposal-class-fields).
```
Comment (klzns) [inline content/docs/typechecking-with-proptypes.md]: Why the link was changed? I think it's better to keep the original one.

Comment (klzns) [inline content/docs/typechecking-with-proptypes.md]: Also, I would change this bit to this:
```
você também poderá declarar `defaultProps` como propriedade estática dentro de uma classe de componente React.
```

File: content/docs/typechecking-with-proptypes.md (hunk 1 above)
Comment (glaucia86) [inline content/docs/typechecking-with-proptypes.md]:
```suggestion
> `React.PropTypes` foi movido para um pacote diferente desde a versão 15.5 do React. Por favor, use [a biblioteca `prop-types`](https://www.npmjs.com/package/prop-types).
```

File: content/docs/typechecking-with-proptypes.md (hunk 3 above)
Comment (lucasdeassis) [reply content/docs/typechecking-with-proptypes.md]: the link ` https://babeljs.io/docs/plugins/transform-class-properties/` redirects to the `https://babeljs.io/docs/en/babel-plugin-proposal-class-properties` so I thought that would be better to change to the new one

File: content/docs/typechecking-with-proptypes.md [hunk 4]
```diff
@@ -143,9 +143,9 @@ MyComponent.propTypes = {
 };
 ```
 
-### Default Prop Values {#default-prop-values}
+### Valores Default de Props {#default-prop-values}
```
Comment (cezaraugusto) [inline content/docs/typechecking-with-proptypes.md]: what if we keep with both translated/original versions here? 

```diff
-Valores Default de Props
+Valores Padrão de Props (Default Prop Values)
```

File: content/docs/typechecking-with-proptypes.md [hunk 5]
```diff
@@ -143,9 +143,9 @@ MyComponent.propTypes = {
...
 
-You can define default values for your `props` by assigning to the special `defaultProps` property:
+Você pode definir valores default para suas `props` através da atribuição à propriedade especial `defaultProps`:
```
Comment (cezaraugusto) [inline content/docs/typechecking-with-proptypes.md]:
```diff
-Você pode definir valores default
+Você pode definir valores padrão (default)
```

File: content/docs/typechecking-with-proptypes.md [hunk 6]
```diff
@@ -168,7 +168,7 @@ ReactDOM.render(
 );
 ```
 
-If you are using a Babel transform like [transform-class-properties](https://babeljs.io/docs/plugins/transform-class-properties/) , you can also declare `defaultProps` as static property within a React component class. This syntax has not yet been finalized though and will require a compilation step to work within a browser. For more information, see the [class fields proposal](https://github.com/tc39/proposal-class-fields).
+Se você está usando um plugin Babel como [plugin-proposal-class-properties](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties), você também poderá declarar `defaultProps` como propriedade estática dentro de uma classe de componente React. Essa sintaxe ainda não foi finalizada e irá exigir uma etapa de compilação para funcionar em um navegador. Para mais informações, veja [proposal-class-fields](https://github.com/tc39/proposal-class-fields).
```
Comment (cezaraugusto) [inline content/docs/typechecking-with-proptypes.md]: I think it's better to keep the original proposal name `proposal-class-fields`

File: content/docs/typechecking-with-proptypes.md [hunk 7]
```diff
@@ -1,18 +1,18 @@
 ---
 id: typechecking-with-proptypes
-title: Typechecking With PropTypes
+title: Checagem de tipos com PropTypes
```
Comment (cezaraugusto) [inline content/docs/typechecking-with-proptypes.md]: can we keep "Typechecking" in the translated version? it seems a very common term. something like the below, for example

```
Checagem de tipos (Typechecking) com PropTypes
```

File: content/docs/typechecking-with-proptypes.md (hunk 4 above)
Comment (lucasdeassis) [reply content/docs/typechecking-with-proptypes.md]: got it. I was trying to figure out a good translation for `default` and couldn't find one. I kept the `default` word but as you mentioned, translating and putting on parenthesis seem better. When using parenthesis on Crowdin I got a warning

File: content/docs/typechecking-with-proptypes.md (hunk 6 above)
Comment (lucasdeassis) [reply content/docs/typechecking-with-proptypes.md]: where in this line?

File: content/docs/typechecking-with-proptypes.md (hunk 2 above)
Comment (halian-vilela) [reply content/docs/typechecking-with-proptypes.md]: This raised me a doubt, which probably will make me change my own translation, hahaha... when the text say **app** should we consider "aplicação" or "aplicativo"?

I think "aplicação" is more general, what do you think?

File: content/docs/typechecking-with-proptypes.md [hunk 8]
```diff
@@ -184,4 +184,4 @@ class Greeting extends React.Component {
 }
 ```
 
-The `defaultProps` will be used to ensure that `this.props.name` will have a value if it was not specified by the parent component. The `propTypes` typechecking happens after `defaultProps` are resolved, so typechecking will also apply to the `defaultProps`.
+O `defaultProps` será usado para garantir que `this.props.name` tenha um valor se não foi especificado pelo componente pai. A checagem de tipos de `propTypes` acontece após `defaultProps` ser resolvido, logo também será aplicada ao `defaultProps`.
```
Comment (halian-vilela) [inline content/docs/typechecking-with-proptypes.md]:
```suggestion
A `defaultProps` será usadas para garantir que `this.props.name` tenha um valor caso não tenha sido especificado pelo componente pai. A checagem de tipos de `propTypes` acontece após `defaultProps` ser resolvida, logo a checagem também será aplicada à `defaultProps`.
```
Making just a little bit clearer.

File: content/docs/typechecking-with-proptypes.md (hunk 6 above)
Comment (fjoshuajr) [inline content/docs/typechecking-with-proptypes.md]:
```suggestion
Se você está usando um plugin Babel como [plugin-proposal-class-properties](https://babeljs.io/docs/plugins/transform-class-properties), você também poderá declarar `defaultProps` como propriedade estática dentro de uma classe de componente React. Essa sintaxe ainda não foi finalizada e irá exigir uma etapa de compilação para funcionar em um navegador. Para mais informações, veja [proposal-class-fields](https://github.com/tc39/proposal-class-fields).
```

Comment (lucasdeassis) [reply content/docs/typechecking-with-proptypes.md]: the url `https://babeljs.io/docs/plugins/transform-class-properties` automatically redirects to `https://babeljs.io/docs/en/babel-plugin-proposal-class-properties`

File: content/docs/typechecking-with-proptypes.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lucasdeassis) [reply content/docs/typechecking-with-proptypes.md]: the verb `atribuir` asks for a complement (`bitransitivo`). Then, if we have `atribuir` we need a target "object". That's why I added the "crase" to it (artigo + preposição).

In `em especial`, I understood that `special propTypes property` mean a particular property, more to the translation of  `very one`" than to the specific word `special`.

[the remark this answers is not in the document]
Comment (lucasdeassis) [reply content/docs/typechecking-with-proptypes.md]: but is really a trick thing. Let me know if you have another POV

File: content/docs/typechecking-with-proptypes.md (hunk 6 above)
Comment (fjoshuajr) [reply content/docs/typechecking-with-proptypes.md]: Ok

**Comment by fjoshuajr** [comment]
> LGTM thanks @lucasdeassis!

I'll merge this then.

**Pull request merged**